crates/router/src/core/payments/helpers.rs

```diff
@@ -3153,14 +3154,49 @@ pub async fn get_merchant_connector_account(
     let db = &*state.store;
     match creds_identifier {
         Some(creds_identifier) => {
-            let mca_config = db
-                .find_config_by_key(format!("mcd_{merchant_id}_{creds_identifier}").as_str())
+            let key = format!("mcd_{merchant_id}_{creds_identifier}");
+            let redis_fetch = || async {
+                db.get_redis_conn()
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("Failed to get redis connection")
+                    .async_and_then(|redis| async move {
+                        redis
+                            .get_and_deserialize_key(key.as_str(), "String")
+                            .await
+                            .change_context(
+                                errors::ApiErrorResponse::MerchantConnectorAccountNotFound {
+                                    id: key,
+                                },
+                            )
+                            .attach_printable("Failed to get redis Value")
```

Comment:
can you also specify which data we failed to retrieve from redis, just for better tracking since `MerchantConnectorAccountNotFound ` does not specify that it is a value from external system

Reply:
does the result from this function logged or returned somewhere?